pkg/db/factory.go

```diff
@@ -72,7 +91,15 @@ func (f *Factory) NewDBStrategy(obj types.Object) (strategy.CompleteStrategy, er
 			tableName = tn.TableName()
 		}
 		if f.AutoMigrate {
-			if err := gdb.Table(tableName).AutoMigrate(&Record{}); err != nil {
+			ctx := context.Background()
+			if f.migrationTimeout != nil {
```

Comment:
nit: since `time.Duration` is really an `int64` under the hood, can we use `f.migrationTimeout != 0` here instead of using a pointer? This may save a bug where the caller may accidentally set a timeout to `0`.

Reply:
I think that's a fair suggestion. I'll update it.

Reply:
done!